moving_modules.md

```diff
@@ -0,0 +1,224 @@
+Moving Non-Core Types & Providers
+=================
+
+## Index
+
+* [Goals](#goals)
+* [User Stories](#user-stories)
+* [File Path Changes](#file-path-changes)
+* [Resource Type Taxonomy](#resource-type-taxonomy)
+  * [Internal](#internal)
+  * [Core](#core)
+  * [External](#external)
+  * [Dependencies](#dependencies)
+* [Packaging](#packaging)
+
+# Goals
+
+* Remove non-core types and providers from the puppet repo and move them to modules. Doing so will reduce the surface area of puppet, decrease puppet CI cycle times, while making the extracted types and providers more accessible to community members and increasing their maintainability.
+* Continue to bundle some modules with puppet-agent during packaging so that users have a batteries-included experience.
+
+# User Stories
+
+## Serverless Puppet
+
+As an admin, I want to run masterless puppet and immediately be able to manage basic resources on my system, so that I can more quickly get value from puppet. I do not want to search the forge in order to perform basic system tasks. The preinstalled modules should make sense for the local platform I'm running puppet on. On *nix, I should be able to manage cron, and on Windows, powershell resources.
```

Comment:
don't know if it matters in the real renderer, but this unmatched, unescaped `*` causes the whole file to be italicised in github's markdown display :(

Reply:
Probably depends on the markdown engine used. We should escape to be safe